wagl/modtran.py

```diff
@@ -11,32 +11,49 @@
 from os.path import join as pjoin, exists, dirname
 import subprocess
 import glob
-
 from posixpath import join as ppjoin
 import numpy
-from scipy.io import FortranFile
 import h5py
 import pandas as pd
-
+import json
+import fnmatch
 from wagl.constants import Workflow, BandType, DatasetName, GroupName, Albedos
 from wagl.constants import POINT_FMT, ALBEDO_FMT, POINT_ALBEDO_FMT
 from wagl.constants import AtmosphericCoefficients as AC
 from wagl.hdf5 import write_dataframe, read_h5_table, create_external_link
 from wagl.hdf5 import VLEN_STRING, write_scalar, H5CompressionFilter
-from wagl.modtran_profiles import MIDLAT_SUMMER_ALBEDO, TROPICAL_ALBEDO
-from wagl.modtran_profiles import MIDLAT_SUMMER_TRANSMITTANCE, SBT_FORMAT
-from wagl.modtran_profiles import TROPICAL_TRANSMITTANCE, THERMAL_TRANSMITTANCE
+import wagl.modtran_profile_json as mpjson
+
 
+class JsonEncoder(json.JSONEncoder):
+    """
+    A wrapper class to address the issue of json encoding error
+    This class handles  the json serializing error for numpy 
+    datatype: 'float32' and numpy arrays
+    """
+    def default(self, obj):
+        if isinstance(obj, numpy.integer):
+            return int(obj)
+        elif isinstance(obj, numpy.floating):
+            return float(obj)
+        elif isinstance(obj, numpy.ndarray):
+            return obj.tolist()
+        else:
+            raise OSError('Json Encoding Error')
 
 def prepare_modtran(acquisitions, coordinate, albedos, basedir, modtran_exe):
     """
     Prepares the working directory for a MODTRAN execution.
     """
-    data_dir = pjoin(dirname(modtran_exe), 'DATA')
+
+    
+    data_dir = pjoin('/g/data/v10/private/modules/MODTRAN/MODTRAN-6.0.1/MODTRAN6.0','DATA')
```

Comment:
Don't include the pathname.  This will need to passed through from the command line.

Reply:
this had been addressed as per Alex's suggestion